sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/hints.scala

```diff
@@ -25,7 +25,7 @@ import org.apache.spark.sql.internal.SQLConf
  * should be removed This node will be eliminated post analysis.
  * A pair of (name, parameters).
  */
-case class UnresolvedHint(name: String, parameters: Seq[String], child: LogicalPlan)
+case class UnresolvedHint(name: String, parameters: Seq[Any], child: LogicalPlan)
```

Comment:
One useful hint parameter is a list of columns.
Something like `df.hint("hint", $"table", Seq($"col1", $"col2", $"col3"))`

In this case UnresolvedHint could be called like this:
```UnresolvedHint(name = String, parameters = Seq(Expression, Seq[Expression]), child)```

But if `UnresolvedHint.parameters` is `Seq[Expression]` then it's not possible to have this kind of hint.